src/main.rs

```diff
@@ -6,50 +6,44 @@
 // copied, modified, or distributed except according to those terms.
 
 extern crate cargo_metadata;
-extern crate docopt;
+extern crate clap;
 extern crate rustc_serialize;
 
 use cargo_metadata::{metadata, Package};
-use docopt::Docopt;
-use std::{env, error, fs, io, path, process};
+use clap::{App, Arg};
+use std::{convert, env, error, fmt, fs, io, path, process};
 use std::io::Write;
 
-const USAGE: &'static str = "
-Cargo Fuzz
-
-Usage:
-  cargo fuzz --init
-  cargo fuzz --fuzz-target TARGET
-  cargo fuzz --add TARGET
-  cargo fuzz --list
-  cargo fuzz (-h | --help)
-
-Options:
-  -h --help              Show this screen.
-  --init                 Initialize fuzz folder
-  --fuzz-target TARGET   Run with given fuzz target in fuzz/fuzzers
-  --add TARGET           Add a new fuzz target
-  --list                 List the available fuzz targets
-";
-
-#[derive(Debug, RustcDecodable)]
-struct Args {
-    flag_init: bool,
-    flag_add: Option<String>,
-    flag_fuzz_target: Option<String>,
-    flag_list: bool,
-}
+const VERSION: Option<&'static str> = option_env!("CARGO_PKG_VERSION");
 
 fn main() {
-    let args: Args = Docopt::new(USAGE)
-                            .and_then(|d| d.decode())
-                            .unwrap_or_else(|e| e.exit());
-
-    let result = if args.flag_init {
+    let app = App::new("cargo-fuzz")
+        .version(VERSION.unwrap_or("unknown"))
+        .arg(Arg::with_name("init")
+            .long("init")
+            .help("Initialize fuzz folder"))
+        .arg(Arg::with_name("fuzz-target")
+            .long("fuzz-target")
+            .value_name("TARGET")
+            .help("Run with given fuzz target in fuzz/fuzzers")
+            .takes_value(true))
+        .arg(Arg::with_name("add")
+            .long("add")
+            .value_name("TARGET")
+            .help("Add a new fuzz target")
+            .takes_value(true))
+        .arg(Arg::with_name("list")
+            .long("list")
+             .help("List the available fuzz targets"));
+    let mut app_help = Vec::new();
+    app.write_help(&mut app_help).ok().expect("could not write help");
+    let args = app.get_matches();
+
+    let result = if args.is_present("init") {
```

Comment:
Should *either* refactor this a bit so something like `cargo fuzz --init --add=name` works (i.e. not if-else chain) or change the CLI a bit to either make the flags exclusive or make the flags subcommands (its fine to break). 

I’d personally prefer changing it, so we’d have something like:

```
cargo fuzz init
cargo fuzz add NAME --maybe --some --options
cargo fuzz target NAME --some --rustc --options -- fuzz options
```